### PR TITLE
Complete ChannelTrace in channelz.proto

### PR DESCRIPTION
**Pull request by dapengzhang0**
Sourced from https://github.com/grpc/proposal/blob/master/A3-channel-tracing.md#format-of-exported-data

### CODE REVIEW

**Comment by ncteisen** [comment]
No this is ready to go, I have one comment though

File: grpc/channelz/v1/channelz.proto [hunk 1]
```diff
@@ -117,10 +117,42 @@ message ChannelData {
   google.protobuf.Timestamp last_call_started_timestamp = 7;
 }
 
+// A trace event is an interesting thing that happened to a channel or
+// subchannel, such as creation, address resolution, subchannel creation, etc.
+message ChannelTraceEvent {
+  // High level description of the event.
+  string description = 1;
+  // The supported severity levels of trace events.
+  enum Severity {
```
Comment (ncteisen) [inline grpc/channelz/v1/channelz.proto]: Can you prefix these with `CT_`? I had to do that to get C++ proto to compile, and I forgot to make the change in my design

Comment (ncteisen) [reply grpc/channelz/v1/channelz.proto]:
```
  enum Severity {
    CT_UNKNOWN = 0;
    CT_INFO = 1;
    CT_WARNING = 2;
    CT_ERROR = 3;
  }
```

**Comment by dapengzhang0** [comment]
C has been using it 
https://github.com/grpc/grpc/pull/13883/files#diff-1d6aca43febb41f16ad4a8dc66ad4b09

File: grpc/channelz/v1/channelz.proto (hunk 1 above)
Comment (ncteisen) [reply grpc/channelz/v1/channelz.proto]: This is the proto that is being used in C++ test; [here](https://github.com/grpc/grpc/blob/master/src/proto/grpc/channelz/channelz.proto). I have been meaning to write a linter to make sure it stays aligned with this one

Comment (dapengzhang0) [reply grpc/channelz/v1/channelz.proto]: Done.

File: grpc/channelz/v1/channelz.proto [hunk 2]
```diff
@@ -117,10 +117,42 @@ message ChannelData {
...
 // ChannelTrace represents the recent events that have occurred on the channel.
 message ChannelTrace {
-  // see the proto in the gRFC for channel tracing:
-  // A3-channel-tracing.md
+  // Number of events ever logged in this tracing object. This can differ from
+  // events.size() because events can be overwritten or garbage collected by
+  // implementations.
+  int64 num_events_logged = 1;
+  // Time that this channel was created.
+  google.protobuf.Timestamp creation_time = 2;
```
Comment (carl-mastrangelo) [inline grpc/channelz/v1/channelz.proto]: Most other time references are name "_timestamp"

Comment (dapengzhang0) [reply grpc/channelz/v1/channelz.proto]: I don't think that's a problem, because the field has a clear description and won't be confuse with time span.

Comment (carl-mastrangelo) [reply grpc/channelz/v1/channelz.proto]: The goal is consistency with the rest of the proto

Comment (dapengzhang0) [reply grpc/channelz/v1/channelz.proto]: Done. I don't mind to rename it, but c++ code will have to refactor after sync.

**Pull request merged**